### ISSUE TITLE
Loopie upgrade TODOs

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#1048]
Let's use this to list the things we'll need to do when we upgrade Loopie to the latest neonFORGE:

- [ ] The **Neon.Kube.Service** package and its namespace have been renamed to **Neon.Service** and the service base class is now named `NeonService`.

- [ ] The `serviceMap` parameter for the `NeonService` constructor has moved to the end of the parameter list and is now optional.  I recommend that Loopie put this after any other parameters and prefix it by `serviceMap:`

- [ ] `KubeService.SetRunning()` has been changed to be `async Task SetRunningAsync()` for `NeonService`.

- [ ] Assign unique ports to the Prometheus metrics ports for the service descriptions in the service map to avoid port conflicts when emulating the cluster:

    ```
    // Assign unique ports to all of the service Prometheus exporters.

    foreach (var serviceDescription in clonedServiceMap.Values)
    {
        serviceDescription.MetricsPort = nextPort++;
    }
    ```
- [ ] Neon test fixtures now support Docker/WSL2 as well as Hyper-V.  We suggest that you switch to WSL2 for better performance.

- [ ] Model generation has changed significantly.  In the past, we bundled modelgen functionality into the **neon-cli**.  That was just an expedient way to deliver this functionality to Loopie developers via neonDESKTOP setup, which included this.  But, it never really made sense to deliver modelgen this way long-term because the neonDESKTOP and neonLIBRARY release cycles have diverged.

    We've added the **Neon.ModelGenerator** package to deliver this functionality going forward.  This includes **linux-x64**, **osx-x64** and **win-x64** versions of the new **neon-modelgen** client which accepts the same command line arguments and options as the old `neon-cli generate model ...` command.  The idea is to reference the **Neon.ModelGenerator** package in your assembly and then follow the instructions in the **Neon.ModelGenerator.readme.txt** file that will be added to your project.

    The nice thing with the new approach is that every project can reference a different version of the model generator and that you can simply use nuget to change to a different version.  With the old way, developers were required by default to use the same **neon-cli** version across all projects and solutions on the workstation which can become problematic for developers working on multiple solutions in parallel.

**Comment by marcusbooyah** [comment]
Closing - moved this to a new issue in the Loopie repo.
